src/dmd/traits.d

```diff
@@ -1810,6 +1811,41 @@ Lnext:
         }
         return r.expressionSemantic(sc);
     }
+    if (e.ident == Id.getLoc)
+    {
+        //Symbol as far as we're concered here can be:
+        // a type,
+        // a dsymbol,
+        // an expression of type DsymbolExp
+
+        if (dim != 1)
+            return dimError(1);
+        auto arg0 = (*e.args)[0];
+        Dsymbol s;
+        {
+            auto t = isType(arg0);
+            auto exp = isExpression(arg0);
+            if (t)
+                s = t.isAggregate();
+            if (exp)
+            {
+                if (auto dse = exp.isDsymbolExp())
+                    s = dse.s;
+            }
+        }
+        if (!s)
+        {
+            e.error("can only get the location of a symbol, not `%s`", arg0.toChars());
+            return new ErrorExp();
+        }
+
+        auto exps = new Expressions(3);
+        exps.data[0] = new StringExp(e.loc, cast(char*)s.loc.filename);
+        exps.data[1] = new IntegerExp(e.loc, s.loc.linnum,Type.tuns32);
```

Comment:
__LINE__ is typed as a tint32.

Reply:
`loc.linnum` is a uint though. Which should it be?

Reply:
The same type as `__LINE__` is, which is `int`.

Reply:
Should it be `int` for the next line as well? Where it includes the column number.